docs/vocabulary.md

```diff
@@ -388,6 +389,230 @@ pipeline in corresponding documentation.
 | **Tekton**         | N/A       | Step       | Task     | Pipeline | Trigger | Resource (?)           |
 | **Zuul**           | N/A       | N/A        | Job      | Pipeline | Trigger | Node (?)               |
 
+### Pipeline Step Types
+
+The table below is an attempt to create a mapping of common names for the various types of Pipeline Steps (the unit of work two degrees smaller than a Pipeline). (The effort of proposing a common vocabulary across CI/CD Tools and Technologies is not yet performed, but we'll use the most common term from the table above as the working term.)
+
+While CI/CD tools and technologies generally give developers broad leeway in naming and implementing their Pipeline Steps, agreeing upon some common terms will help develop more pluggable pipelines. This list is not meant to be exhaustive.
+
+Each type of Pipeline Step will generally have several implementations that are tool specific. For example, Source will have implementations for various Source Code Management (SCM) tools. Publish will have implementations for each type of repository, and so on.
+
+#### Setup
+* Semantics: Provision build resources, set up the build workspace.
```

Comment:
i had a similar comment on PR https://github.com/cdfoundation/sig-interoperability/pull/76#discussion_r788875879 since similar setup steps are required for other stages.

Reply:
Let's put "Build Setup" and "Test Setup" as two different examples of setup steps.  If you're running one pipeline that builds and tests you might only need one setup step, or you might want a "daily test pipeline" that only setups up its test environment and runs tests, without a build.

Reply:
Sounds great!

Reply:
I'll put this in a separate PR just because the merge conflicts are getting fun.

Reply:
Actually, nix that last comment; I had already taken "build" out of the name.